Linker generator broken for nRF devices

**Describe the bug**
The following PR https://github.com/zephyrproject-rtos/zephyr/pull/71590
 broke support for `CONFIG_CMAKE_LINKER_GENERATOR=y` and thereby also broke support for [ARM Compiler 6](https://docs.zephyrproject.org/latest/develop/toolchains/arm_compiler.html) for all Nordic nRF devices.

**To Reproduce**
Steps to reproduce the behavior:
1. cmake -GNinja -Bbuild -DBOARD=nrf52840dk/nrf52840 samples/hello_world/ -DCONFIG_CMAKE_LINKER_GENERATOR=y
2. ninja -Cbuild
3. See error

```
/opt/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/arch/arch/arm/core/cortex_m/libarch__arm__core__cortex_m.a(reset.S.obj): in function `__start':
/projects/github/ncs/zephyr/arch/arm/core/cortex_m/reset.S:97: undefined reference to `z_arm_platform_init'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Expected behavior**
A successful build

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)

--- a/soc/nordic/common/CMakeLists.txt
+++ b/soc/nordic/common/CMakeLists.txt
@@ -3,7 +3,13 @@
 
 add_subdirectory_ifdef(CONFIG_RISCV_CORE_NORDIC_VPR vpr)
 
-zephyr_linker_sources_ifdef(CONFIG_ARM SECTIONS arm_platform_init.ld)
+if(CONFIG_ARM)
+  # Let SystemInit() be called in place of z_arm_platform_init() by default.
+  zephyr_linker_symbol(SYMBOL z_arm_platform_init EXPR "@SystemInit@")
+
+  # This file is used when the CMake linker script generator is disabled.
+  zephyr_linker_sources(SECTIONS arm_platform_init.ld)
+endif()
 
 zephyr_library_sources_ifdef(CONFIG_POWEROFF poweroff.c)
 

